### ISSUE TITLE
Navigation broken when deployed in Netlify

### ISSUE THREAD

**Issue by EmaSuriano** [narative/gatsby-theme-novela#216]
**Version of Novela you are using**
0.10.3

**Number of posts you have and which data source**
Local

**Describe the bug**
It's not directly connected with this theme but it might be because I'm reusing one of the components inside the theme.

I created a new `index.js` that replaces the original one, in which I'm rendering the same `Articles.List`.  The problem I'm having is when I deployed the site to Netlify, whenever I want to go to any of this article I'm having this error inside the console.

```
TypeError: c is not a function
    at d2b98e9cd65bee24f1e718c407c08d2562fa1975.js:1
    at KqkS.exports.unstable_runWithPriority (d2b98e9cd65bee24f1e718c407c08d2562fa1975.js:1)
    at fg (d2b98e9cd65bee24f1e718c407c08d2562fa1975.js:1)
    at Vi (d2b98e9cd65bee24f1e718c407c08d2562fa1975.js:1)
    at Xi (d2b98e9cd65bee24f1e718c407c08d2562fa1975.js:1)
    at ik (d2b98e9cd65bee24f1e718c407c08d2562fa1975.js:1)
    at KqkS.exports.unstable_runWithPriority (d2b98e9cd65bee24f1e718c407c08d2562fa1975.js:1)
    at fg (d2b98e9cd65bee24f1e718c407c08d2562fa1975.js:1)
    at Yj (d2b98e9cd65bee24f1e718c407c08d2562fa1975.js:1)
    at Lj (d2b98e9cd65bee24f1e718c407c08d2562fa1975.js:1)
```

I tried to run `gatsby build` and `gatsby serve` and I can't reproduce this issue locally :/ 

**To Reproduce**
1. Go to https://emasuriano-new.netlify.com/
2. Click on any of the articles inside the `Latest Articles`
3. The page goes blank
4. But if you refresh the page, now it works ...

**Expected behavior**
Navigation should work without the need to refresh.

**Browser**
 - Chrome

**Comment by EmaSuriano** [comment]
This is the list of plugins I'm using:

``` javascript
 plugins: [
    'gatsby-plugin-netlify-cache',
    'gatsby-plugin-netlify',
    {
      resolve: '@narative/gatsby-theme-novela',
      options: {
        contentPosts: 'content/posts',
        contentAuthors: 'content/about',
        basePath: '/',
        authorsPage: false,
        mailchimp: true,
        sources: {
          local: true,
        },
      },
    },
    {
      resolve: 'gatsby-plugin-mailchimp',
      options: {
        endpoint:
          'https://gmail.us20.list-manage.com/subscribe/post?u=xxxxxxxxxx&amp;id=fcd9f3d1f5',
      },
    },
    {
      resolve: 'gatsby-plugin-manifest',
      options: {
        name: 'test',
        short_name: 'test',
        start_url: '/',
        background_color: '#fff',
        theme_color: '#fff',
        display: 'standalone',
        icon: 'src/assets/favicon.png',
      },
    },
  ],
```

And these are my dependencies:

``` json
"dependencies": {
    "@emotion/core": "^10.0.22",
    "@emotion/styled": "^10.0.23",
    "@narative/gatsby-theme-novela": "^0.*",
    "gatsby": "^2.15.7",
    "gatsby-plugin-mailchimp": "^5.1.2",
    "gatsby-plugin-manifest": "^2.2.4",
    "gatsby-plugin-netlify": "^2.1.25",
    "gatsby-plugin-netlify-cache": "^1.2.0",
    "js-yaml": "^3.13.1",
    "react": "^16.8.6",
    "react-dom": "^16.8.6",
    "react-typed": "^1.2.0"
  },
```

**Comment by brotzky** [comment]
First off, looks great!

Secondly, it looks like the link works when going directly from https://emasuriano-new.netlify.com/blog.

It's hard to identify the root of the problem, but it seems you're on the right path. Sorry, I can't help much beyond that!

**Comment by EmaSuriano** [comment]
@brotzky yeah that's also something I discovered it too. Are we creating the pages inside the theme in a "special" or loading them dynamically in some part of the app?

**Comment by brotzky** [comment]
No, it should all be within createPages!

**Comment by EmaSuriano** [comment]
So after doing some test I finally discovered the issue. The problem relays on `Article.List` is using the `Link` component from `gatsby`. This for some reason has some problem regarding the caching inside Netlify which ends up on an undefined problem inside the console after being deployed.

The way that I fixed it was by using the shadowing feature of themes built in gatsby and replacing the `styled(Link)` with `styled.a` in this line --> https://github.com/narative/gatsby-theme-novela/blob/master/@narative/gatsby-theme-novela/src/sections/articles/Articles.List.tsx#L343

I would say this issue can be closed because it's not directly related to the theme 😄
